From 965c4531505817056ae92ae7d1f829b6b02f0c13 Mon Sep 17 00:00:00 2001
From: Mahesh Kumar P <mahesh.kumar.p@intel.com>
Date: Thu, 21 May 2015 11:52:01 +0530
Subject: [PATCH] cpufreq: Fix the shared policy setting for CHT platforms

The patch fixes the shared cpufreq policy for CHT platforms.
The patch is made generic to CHT platforms. On T3 , all cores
should run at the same frequencies, whereas on T4, the modules
should run at same frequencies.

Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-9817
Change-Id: I2787b5d78b7c8d44d095402ef5cb3775fa0a0d48
Signed-off-by: Kumar P, Mahesh <mahesh.kumar.p@intel.com>
---
 drivers/cpufreq/acpi-cpufreq.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/cpufreq/acpi-cpufreq.c b/drivers/cpufreq/acpi-cpufreq.c
index fa9b191b942f..cfe9b3d63523 100644
--- a/drivers/cpufreq/acpi-cpufreq.c
+++ b/drivers/cpufreq/acpi-cpufreq.c
@@ -731,11 +731,14 @@ static int acpi_cpufreq_cpu_init(struct cpufreq_policy *policy)
 		intel_mid_msgbus_read32(CLASSCODE_REG, CLASSCODE_OFFSET);
 		revision_id = (revision_id & REVISION_ID_BIT);
 
-		if ((c->x86_model == 0x4c) && (!revision_id)) {
-			pr_info("Platform supports Module Level DVFS\n");
+		if (c->x86_model == 0x4c) {
 			policy->shared_type = CPUFREQ_SHARED_TYPE_ALL;
-			get_cpu_sibling_mask(cpu, &sibling_mask);
-			cpumask_copy(policy->cpus, &sibling_mask);
+			cpumask_copy(policy->cpus, perf->shared_cpu_map);
+			if (!revision_id) {
+				pr_info("Platform supports Module Level DVFS\n");
+				get_cpu_sibling_mask(cpu, &sibling_mask);
+				cpumask_copy(policy->cpus, &sibling_mask);
+			}
 		}
 	}
 #endif
-- 
1.9.1

